### PR TITLE
Add SwiftHash again

### PR DESCRIPTION
**Pull request by onmyway133**
- So because @vadymmarkov & @zenangst like `SwiftHash` so much, it is back to review again 😉 
- There 's problem with `has been explicitly marked unavailable` on `macOS` target, but that should be fixed in another PR
- Also, we should add CI for this, but that should be in another PR

### CODE REVIEW

File: Cache.podspec [hunk 1]
```diff
@@ -19,7 +19,7 @@ Pod::Spec.new do |s|
   s.tvos.source_files = 'Source/{iOS,Shared}/**/*'
 
   s.frameworks = 'Foundation'
-  s.dependency 'CryptoSwift'
+  s.dependency 'SwiftHash'
```
Comment (vadymmarkov) [inline Cache.podspec]: @onmyway133 Can you specify a version?

File: Source/Shared/Storage/DiskStorage.swift [hunk 2]
```diff
@@ -307,16 +307,7 @@ public final class DiskStorage: StorageAware {
...
   func fileName(_ key: String) -> String {
-    if let digest = key.data(using: String.Encoding.utf8)?.md5() {
-      var string = ""
-      for byte in digest {
-        string += String(format:"%02x", byte)
-      }
-
-      return string
-    } else {
-      return key.base64()
```
Comment (vadymmarkov) [inline Source/Shared/Storage/DiskStorage.swift]: Can you remove `base64` method. I don't think we use it anymore.

Comment (onmyway133) [inline Source/Shared/Storage/DiskStorage.swift]: Yeah, I will. This if/else with `md5` and `base64` will make it inconsistent. Better to stick with 1 only

**Comment by onmyway133** [comment]
Yeah 🕺 💃 👯‍♂️ 👯

**Pull request merged**